### PR TITLE
Projectiles Dev Tool and Additions to the AoE plugin

### PR DESCRIPTION
**Pull request by Dreyri**
An example of what the Projectile Dev Tool looks like

ss 2018-01-14 at 04 14 14

Additionally the following were added to the AoE plugin.
- Vorkath
- Galvek
- Vet'ion
- Chaos Fanatic
- Corporeal Beast

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java [hunk 1]
```diff
@@ -325,6 +314,80 @@ private void renderInventory(Graphics2D graphics)
...
+			}
+			else
+			{
+				net.runelite.api.Point targetPoint = projectile.getTargetPoint();
+				OverlayUtil.renderTilePointOverlay(graphics, client, targetPoint, infoString, Color.RED);
+			}
+
+		}
+
+		/*
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: Never commit commented out code

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java [hunk 2]
```diff
@@ -25,30 +25,14 @@
  */
 package net.runelite.client.plugins.devtools;
 
-import java.awt.Color;
-import java.awt.Dimension;
-import java.awt.Font;
-import java.awt.FontMetrics;
-import java.awt.Graphics2D;
+import java.awt.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: Using * for imports is usually discouraged (at least in Java standards) as it reduces explicity of the imports

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java [hunk 3]
```diff
@@ -25,30 +25,14 @@
...
-import net.runelite.api.Item;
-import net.runelite.api.ItemLayer;
-import net.runelite.api.NPC;
-import net.runelite.api.Node;
-import net.runelite.api.Player;
-import net.runelite.api.Region;
-import net.runelite.api.Tile;
-import net.runelite.api.WallObject;
+
+import net.runelite.api.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: See my previous comment

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java [hunk 4]
```diff
@@ -325,6 +314,80 @@ private void renderInventory(Graphics2D graphics)
 		}
 	}
 
+	public void renderProjectiles(Graphics2D graphics)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: This method can be private

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java [hunk 5]
```diff
@@ -325,6 +314,80 @@ private void renderInventory(Graphics2D graphics)
...
+		/*
+		for(Projectile proj : projectiles)
+		{
+			//proj.getId();
+			System.out.println("Projectile id: " + proj.getId());
+		}
+		*/
+	}
+
+	public void renderProjectileOrigin(Graphics2D graphics, Projectile projectile, int floor, net.runelite.api.Point origin)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: And this one can be private as well

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayUtil.java [hunk 6]
```diff
@@ -37,10 +37,8 @@
 import java.awt.RenderingHints;
 import java.awt.geom.Point2D;
 import java.awt.image.BufferedImage;
-import net.runelite.api.Actor;
-import net.runelite.api.Point;
-import net.runelite.api.SpritePixels;
-import net.runelite.api.TileObject;
+
+import net.runelite.api.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayUtil.java]: See my previous comment about using *

File: runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java [hunk 7]
```diff
@@ -26,17 +26,8 @@
...
-import net.runelite.api.MenuAction;
-import net.runelite.api.MenuEntry;
-import net.runelite.api.NPC;
-import net.runelite.api.Player;
-import net.runelite.api.Point;
-import net.runelite.api.Prayer;
-import net.runelite.api.Skill;
-import net.runelite.api.Varbits;
+
+import net.runelite.api.*;
```
Comment (deathbeam) [inline runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java]: See my previous comment about using *

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java (hunk 2 above)
Comment (Dreyri) [reply runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: the * is IntelliJs doing, can I disable that somewhere?

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/devtools/DevToolsOverlay.java]: https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports#3348855

**Pull request merged**